compiler/src/test/java/com/squareup/anvil/compiler/InterfaceMergerTest.kt

```diff
@@ -622,4 +623,26 @@ class InterfaceMergerTest(
       assertThat(componentInterface extends secondContributingInterface).isTrue()
     }
   }
+
+  @Test fun `anonymous inner classes should not be evaluated`() {
+    compile(
+      """
+      package com.squareup.test
+
+      class NonContributingClass {
+        private val anonymousDoSomething = object : DoSomethingInterface {
+          override fun doSomething() { 
+            // unimportant impl
+          }
```

Comment:
```suggestion
          override fun doSomething() = Unit
```
We use that to indicate that the body is empty on purpose.